docs/Upgrade.md

```diff
@@ -2121,42 +2123,32 @@ If you relied on the `handleSubmitWithRedirect` to change the redirection:
 ```diff
 const MyForm = () => {
     return (
-        <Create>
-            <FormWithRedirect
+-        <Create>
++        <Create redirect="show">
+-            <FormWithRedirect
++            <Form>
 -                render={({ handleSubmitWithRedirect, ...formProps }) => (
-+                render={({ handleSubmit, ...formProps }) => (
-+                    <form onSubmit={handleSubmit}>
                          <TextInput source="name" />
 -                        <MySaveButton handleSubmitWithRedirect={handleSubmitWithRedirect}>
 +                        <MySaveButton />
-+                    </form>
-                )}
-            />
+-                )}
+-            />
++            </Form>
         </Create>
     );
 };
 
-import { useSaveContext, useRedirect } from 'react-admin';
-import { useForm } from 'react-final-form';
 -const MySaveButton = ({ handleSubmitWithRedirect }) => (
 +const MySaveButton = () => {
-+    const { save } = useSaveContext();
-+    const form = useForm(); 
-+    const redirect = useRedirect();
-+    const handleClick = (event) => {
-+        event.preventDefault(); // Prevent the default form submission
-+        const values = form.getState().values;
-+        save(values, {
-+            onSuccess: (data) => redirect('show', '/posts', data.id)
-+        })
-+    };
     return (
 -        <button onClick={() => handleSubmitWithRedirect('show')}>Save</button>
-+        <button type="button" onClick={handleClick}>Save</button>
++        <button type="submit">Save</button>
     );
 );
 ```
 
+You can also use a function to handle the redirection target, as described in [this section](./EditTutorial.md#redirection-after-submission).
```

Comment:
```suggestion
You can also use a function to handle the redirection target, as described in [the redirection after submittion documentation](./EditTutorial.md#redirection-after-submission).
```

Don't use 'this thing' as support of a hyperlink, but rather the title of the this you link to.